scm-webapp/src/main/java/sonia/scm/search/SearchableTypeResolver.java

```diff
@@ -81,22 +82,22 @@ private void fillMaps(Iterable<LuceneSearchableType> types) {
     return Collections.unmodifiableCollection(classToSearchableType.values());
   }
 
-  public LuceneSearchableType resolve(Object object) {
+  public Optional<LuceneSearchableType> resolve(Object object) {
     return resolve(object.getClass());
   }
 
-  public LuceneSearchableType resolve(Class<?> type) {
+  public Optional<LuceneSearchableType> resolve(Class<?> type) {
     if (type == null) {
-      throw notFound(entity("type", "null"));
+      return Optional.empty();
     }
     LuceneSearchableType searchableType = classToSearchableType.get(type);
     if (searchableType == null) {
       throw notFound(entity("type", type.getName()));
     }
-    return searchableType;
+    return Optional.of(searchableType);
   }
 
-  public LuceneSearchableType resolveByName(String typeName) {
+  public Optional<LuceneSearchableType> resolveByName(String typeName) {
     Class<?> type = nameToClass.get(typeName);
     if (type == null) {
       throw notFound(entity("type", typeName));
```

Comment:
Sorry, but I don't like this. The only reason for the `Optionals` here is, that the second function might be calles with a `null` argument. And this only can happen in the `LuceneForIndices#exec` function. Instead of changing all these APIs with unclear semantics (maybe you will get an empty optional, maybe you will get a not found exception), we should filter these `null` cases and keep the API as it was.

But maybe I'm missing something here?